### PR TITLE
web practice 1

### PR DESCRIPTION
**Pull request by Avassor3000**
Deleting cars is done with link & get method, other deletes are done with button & post method

### CODE REVIEW

**Review by andriiyaroslav — APPROVED** [review]

File: src/main/java/mate/controller/car/AddDriverToCarController.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.model.Car;
+import mate.model.Driver;
+import mate.service.CarService;
+import mate.service.DriverService;
+
+@WebServlet(urlPatterns = "/cars/to")
```
Comment (Nick97-git) [inline src/main/java/mate/controller/car/AddDriverToCarController.java:15]: look at the checklist and fix naming of URLs everywhere

File: src/main/java/mate/controller/car/AddDriverToCarController.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    private static Injector injector = Injector.getInstance("mate");
+    private final CarService carService = (CarService) injector
+            .getInstance(CarService.class);
+    private final DriverService driverService = (DriverService) injector
+            .getInstance(DriverService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/cars/to.jsp")
```
Comment (Nick97-git) [inline src/main/java/mate/controller/car/AddDriverToCarController.java:26]:
```suggestion
        req.getRequestDispatcher("/WEB-INF/views/cars/drivers/add.jsp")
```

File: src/main/java/mate/controller/car/CreateCarController.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/cars/add.jsp")
+                .forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String model = req.getParameter("model");
+        Long id = Long.valueOf(req.getParameter("manufacturerId"));
```
Comment (Nick97-git) [inline src/main/java/mate/controller/car/CreateCarController.java:36]:
```suggestion
        Long manufacturerId = Long.valueOf(req.getParameter("manufacturerId"));
```
taking into account that we create a car here, it's better to specify whose id this is

File: src/main/java/mate/controller/manufacturer/GetAllManufacturerController.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        req.getRequestDispatcher("/WEB-INF/views/manufacturers/all.jsp")
+                .forward(req, resp);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        Long id = Long.valueOf(req.getParameter("manufacturerId"));
+        manufacturerService.delete(id);
+        resp.sendRedirect("/index");
```
Comment (Nick97-git) [inline src/main/java/mate/controller/manufacturer/GetAllManufacturerController.java:32]: delete operation in GetAll?

File: src/main/webapp/WEB-INF/views/cars/add.jsp [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+</h1>
+<form method="post" action="${pageContext.request.contextPath}/cars/add">
+    Car model <input type="text" name="model">
+    <br>
+    Manufacturer id <input type="text" name="manufacturerId">
+    <br>
+    <button type="submit">Submit</button>
+</form>
+</body>
+</html>
```
Comment (Nick97-git) [inline src/main/webapp/WEB-INF/views/cars/add.jsp:19]: add empty line in the end of each jsp

File: src/main/webapp/WEB-INF/views/cars/all.jsp [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+<%--<h3>--%>
+<%--    You can delete driver:--%>
+<%--</h3>--%>
+<%--<form method="post" action="${pageContext.request.contextPath}/drivers/all">--%>
+<%--    Driver Id <input type="text" name="driverId">--%>
+<%--    <br>--%>
+<%--    <button type="submit">Submit</button>--%>
+<%--</form>--%>
+
```
Comment (Nick97-git) [inline src/main/webapp/WEB-INF/views/cars/all.jsp:66]: remove comments

File: src/main/webapp/WEB-INF/views/manufacturers/all.jsp [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+</body>
+</html>
+
+
+
+
+
+
+
+
```
Comment (Nick97-git) [inline src/main/webapp/WEB-INF/views/manufacturers/all.jsp:60]: remove redundant empty lines

File: src/main/webapp/web.xml [hunk 8]
```diff
@@ -3,12 +3,12 @@
...
-        <url-pattern>/index</url-pattern>
-    </servlet-mapping>
+<!--    <servlet>-->
+<!--        <servlet-name>index</servlet-name>-->
+<!--        <servlet-class>mate.controller.IndexController</servlet-class>-->
+<!--    </servlet>-->
+<!--    <servlet-mapping>-->
+<!--        <servlet-name>index</servlet-name>-->
+<!--        <url-pattern>/index</url-pattern>-->
+<!--    </servlet-mapping>-->
```
Comment (Nick97-git) [inline src/main/webapp/web.xml:13]: remove comments

**Review by Nick97-git — CHANGES_REQUESTED** [review]

File: src/main/webapp/web.xml (hunk 8 above)
Comment (Avassor3000) [reply src/main/webapp/web.xml:13]: +

File: src/main/webapp/WEB-INF/views/manufacturers/all.jsp (hunk 7 above)
Comment (Avassor3000) [reply src/main/webapp/WEB-INF/views/manufacturers/all.jsp:60]: +

File: src/main/webapp/WEB-INF/views/cars/all.jsp (hunk 6 above)
Comment (Avassor3000) [reply src/main/webapp/WEB-INF/views/cars/all.jsp:66]: +

File: src/main/webapp/WEB-INF/views/cars/add.jsp (hunk 5 above)
Comment (Avassor3000) [reply src/main/webapp/WEB-INF/views/cars/add.jsp:19]: +

File: src/main/java/mate/controller/car/CreateCarController.java (hunk 3 above)
Comment (Avassor3000) [reply src/main/java/mate/controller/car/CreateCarController.java:36]: +

File: src/main/java/mate/controller/car/AddDriverToCarController.java (hunk 1 above)
Comment (Nick97-git) [reply src/main/java/mate/controller/car/AddDriverToCarController.java]: didn't fix, look at the checklist

File: src/main/java/mate/controller/car/AddDriverToCarController.java (hunk 2 above)
Comment (Nick97-git) [reply src/main/java/mate/controller/car/AddDriverToCarController.java]: didn't fix

File: src/main/java/mate/controller/driver/AddNewDriverController.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+@WebServlet(urlPatterns = "/drivers/all/add")
+public class AddNewDriverController extends HttpServlet {
+    private static Injector injector = Injector.getInstance("mate");
+    private final DriverService driverService = (DriverService) injector
+            .getInstance(DriverService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/drivers/all.jsp")
```
Comment (Nick97-git) [inline src/main/java/mate/controller/driver/AddNewDriverController.java:22]: why did you delete several jsps, you should use in such cases separate jsp for addition of a new car, return jsps, that you deleted

**Review by Nick97-git — CHANGES_REQUESTED** [review]

**Review by Nick97-git — CHANGES_REQUESTED** [review]
you didn't fix two comments from my previous review, they're unresolved + left one new comment

File: src/main/java/mate/controller/car/AddDriverToCarController.java [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.model.Car;
+import mate.model.Driver;
+import mate.service.CarService;
+import mate.service.DriverService;
+
+@WebServlet(urlPatterns = "/cars/driver/add")
```
Comment (Nick97-git) [inline src/main/java/mate/controller/car/AddDriverToCarController.java:15]:
```suggestion
@WebServlet(urlPatterns = "/cars/drivers/add")
```

File: src/main/java/mate/controller/car/AllCarsController.java [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.service.CarService;
+
+@WebServlet(urlPatterns = "/cars/all")
```
Comment (Nick97-git) [inline src/main/java/mate/controller/car/AllCarsController.java:12]:
```suggestion
@WebServlet(urlPatterns = "/cars")
```

File: src/main/java/mate/controller/car/CreateCarController.java [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.model.Car;
+import mate.model.Manufacturer;
+import mate.service.CarService;
+import mate.service.ManufacturerService;
+
+@WebServlet(urlPatterns = "/cars/add")
+public class CreateCarController extends HttpServlet {
```
Comment (Nick97-git) [inline src/main/java/mate/controller/car/CreateCarController.java:17]:
```suggestion
public class AddCarController extends HttpServlet {
```
the same for other similar cases

File: src/main/java/mate/controller/car/DeleteCarController.java [hunk 13]
```diff
@@ -0,0 +1,26 @@
...
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.service.CarService;
+
+@WebServlet(urlPatterns = "/cars/all/del")
```
Comment (Nick97-git) [inline src/main/java/mate/controller/car/DeleteCarController.java:12]:
```suggestion
@WebServlet(urlPatterns = "/cars/delete")
```

File: src/main/java/mate/controller/driver/DeleteDriverController.java [hunk 14]
```diff
@@ -0,0 +1,26 @@
...
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.service.DriverService;
+
+@WebServlet(urlPatterns = "/drivers/all/del")
```
Comment (Nick97-git) [inline src/main/java/mate/controller/driver/DeleteDriverController.java:12]:
```suggestion
@WebServlet(urlPatterns = "/drivers/delete")
```

File: src/main/java/mate/controller/driver/GetAllDriversController.java [hunk 15]
```diff
@@ -0,0 +1,26 @@
...
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.service.DriverService;
+
+@WebServlet(urlPatterns = "/drivers/all")
```
Comment (Nick97-git) [inline src/main/java/mate/controller/driver/GetAllDriversController.java:12]:
```suggestion
@WebServlet(urlPatterns = "/drivers")
```

File: src/main/java/mate/controller/manufacturer/CreateManufacturerController.java [hunk 16]
```diff
@@ -0,0 +1,37 @@
...
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.model.Manufacturer;
+import mate.service.ManufacturerService;
+
+@WebServlet(urlPatterns = "/manufacturers/add")
+public class CreateManufacturerController extends HttpServlet {
```
Comment (Nick97-git) [inline src/main/java/mate/controller/manufacturer/CreateManufacturerController.java:14]:
```suggestion
public class AddManufacturerController extends HttpServlet {
```

File: src/main/java/mate/controller/manufacturer/DeleteManufacturerController.java [hunk 17]
```diff
@@ -0,0 +1,26 @@
...
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.service.ManufacturerService;
+
+@WebServlet(urlPatterns = "/manufacturers/all/del")
```
Comment (Nick97-git) [inline src/main/java/mate/controller/manufacturer/DeleteManufacturerController.java:12]:
```suggestion
@WebServlet(urlPatterns = "/manufacturers/delete")
```

File: src/main/java/mate/controller/manufacturer/GetAllManufacturerController.java [hunk 18]
```diff
@@ -0,0 +1,26 @@
...
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import mate.lib.Injector;
+import mate.service.ManufacturerService;
+
+@WebServlet(urlPatterns = "/manufacturers/all")
```
Comment (Nick97-git) [inline src/main/java/mate/controller/manufacturer/GetAllManufacturerController.java:12]:
```suggestion
@WebServlet(urlPatterns = "/manufacturers")
```

File: src/main/java/mate/controller/car/AddDriverToCarController.java [hunk 19]
```diff
@@ -0,0 +1,40 @@
...
+    private static Injector injector = Injector.getInstance("mate");
+    private final CarService carService = (CarService) injector
+            .getInstance(CarService.class);
+    private final DriverService driverService = (DriverService) injector
+            .getInstance(DriverService.class);
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        req.getRequestDispatcher("/WEB-INF/views/cars/driver/add.jsp")
```
Comment (Nick97-git) [inline src/main/java/mate/controller/car/AddDriverToCarController.java:26]:
```suggestion
        req.getRequestDispatcher("/WEB-INF/views/cars/drivers/add.jsp")
```

**Review by Nick97-git — CHANGES_REQUESTED** [review]

**Review by Nick97-git — APPROVED** [review]
